examples/discovery-search-app/src/App.js

```diff
@@ -75,7 +76,11 @@ const App = () => {
   ) : isError ? (
     <div>Unable to load Discovery projects. Please check your console for more details.</div>
   ) : (
-    <DiscoverySearch searchClient={searchClient} projectId={projectId}>
+    <DiscoverySearch
+      searchClient={searchClient}
+      projectId={projectId}
+      documentProvider={new ExampleDocumentProvider()}
```

Comment:
This is how a user would enable the document provider, so that `DocumentPreview` invokes its methods automatically when wanting to render a document.

The other option is more "manual": simply manage the loading of `DocumentPreview` and pass in the correct `file` when needed.